src/api/Nhs.Appointments.Api/Notifications/ServiceRegistration.cs

```diff
@@ -0,0 +1,54 @@
+﻿using MassTransit;
+using Microsoft.Extensions.DependencyInjection;
+using Nhs.Appointments.Api.Consumers;
+using Nhs.Appointments.Api.Functions;
+using Nhs.Appointments.Core.Messaging;
+using Nhs.Appointments.Core.Messaging.Events;
+using Notify.Interfaces;
+using System;
+
+namespace Nhs.Appointments.Api.Notifications;
+
+public static class ServiceRegistration
+{
+    public static IServiceCollection AddUserNotifications(this IServiceCollection services)
+    {
+        var userNotificationsProvider = Environment.GetEnvironmentVariable("Notifications_Provider");
+
+        services.Configure<UserRolesChangedNotifier.Options>(opts =>
+        {
+            opts.EmailTemplateId = Environment.GetEnvironmentVariable("UserRolesChangedEmailTemplateId");
+        });
+
+        services.AddScoped<IAsyncNotificationClient>(x => new Notify.Client.NotificationClient(Environment.GetEnvironmentVariable("GovNotifyApiKey")));
+   
+        if (userNotificationsProvider == "local")
+        {
+            services
+                .AddTransient<IUserRolesChangedNotifier, UserRolesChangedNotifier>()
+                .AddScoped<IConsumer<UserRolesChanged>, UserRolesChangedConsumer>()
+                .AddScoped<IMessageBus, ConsoleLogWithMessageDelivery>();
```

Comment:
Does this mean that local running now relies on a connection to gov notify? Would it be more appropriate to have that as a separate config setting or is there a way in this configuration to have the actual sending via gov notify mocked out